src/client/kube_store/index.ts

```diff
@@ -233,6 +47,48 @@ export default class KubeStore {
     return this.instance || (this.instance = new this(''));
   }
 
+  public patchResource(resource: KubeResource, name: string, patch: object): object {
+    let result: object = {};
+    let key: string = '';
+
+    if (resource instanceof NamespacedResource) {
+      const namespacedResource = resource as NamespacedResource;
+      if (resource instanceof CoreNamespacedResource) {
+        // Core Namespaced
+        key = `api/${resource.gvk().version}/namespaces/${namespacedResource.namespace}/${resource.gvk().kindPlural}`;
+        if (this.data()[key]) {
+          result = this.data()[key][name];
+        }
+      } else {
+        // Extension Namespaced
+        key = `apis/${resource.gvk().group}/${resource.gvk().version}` +
+          `/namespaces/${namespacedResource.namespace}/${resource.gvk().kindPlural}`;
+        if (this.data()[key]) {
+          result = this.data()[key][name];
+        }
+      }
+    } else {
+      if (resource instanceof CoreClusterResource) {
+        // Core ClusterResource
+        key = `api/${resource.gvk().version}/namespaces/${resource.gvk().kindPlural}`;
+        if (this.data()[key]) {
+          result = this.data()[key][name];
+        }
+      } else {
+        // Extension ClusterResource
+        key = `apis/${resource.gvk().group}/${resource.gvk().version}/${resource.gvk().kindPlural}`;
+        if (this.data()[key]) {
+          result = this.data()[key][name];
+        }
+      }
+    }
+
+    const patchedObject = JsonMergePatch.apply(resource, patch);
```

Comment:
Patching 'resource' is incorrect.
In this context, 'resource' is an object that just tracks the GVK metadata, it does not have the actual data.

(This may be confusing as the name 'Resource' is used in k8s and typically would refer to object data, in this case we are using the class definition 'Resource' to track metadata associated with the GVK.  'Resource' in this context is essentially a class that helps us to identify the URL of the Resource on k8s API)

What you want to do is to apply the json merge patch to the data.

You want to:
1. Extract the desired data (you use the resource and it's gvk methods to form a key so you can lookup the specific data you want from localStorage).  
2. Apply the merge patch to the actual data which is in the store for this specific object

One issue in the method way it is written is that you are not getting the data.

You have a block above which is doing a lookup on this.data()[key][name], and putting the data in 'result'... yet you are not using 'result'.

I would have expected to see:
const patchedObject = JsonMergePatch.apply(result, patch);

Reply:
fixed